File: src/main/java/org/taktik/icure/services/external/rest/v1/dto/IdentityDocumentReaderDto.java
package org.taktik.icure.services.external.rest.v1.dto;

public class IdentityDocumentReaderDto {

    protected String justificatifDocumentNumber;
    protected String SupportSerialNumber;
    protected Long timeReadingEIdDocument;
    protected int eIdDocumentSupportType;
    protected int reasonManualEncoding;
    protected int reasonUsingVignette;

    public String getJustificatifDocumentNumber() {
        return justificatifDocumentNumber;
    }

    public void setJustificatifDocumentNumber(String justificatifDocumentNumber) {
        this.justificatifDocumentNumber = justificatifDocumentNumber;
    }

    public String getSupportSerialNumber() {
        return SupportSerialNumber;
    }

    public void setSupportSerialNumber(String supportSerialNumber) {
        SupportSerialNumber = supportSerialNumber;
    }

    public Long getTimeReadingEIdDocument() {
        return timeReadingEIdDocument;
    }

    public void setTimeReadingEIdDocument(Long timeReadingEIdDocument) {
        this.timeReadingEIdDocument = timeReadingEIdDocument;
    }

    public int geteIdDocumentSupportType() {
        return eIdDocumentSupportType;
    }

    public void seteIdDocumentSupportType(int eIdDocumentSupportType) {
        this.eIdDocumentSupportType = eIdDocumentSupportType;
    }

    public int getReasonManualEncoding() {
        return reasonManualEncoding;
    }

    public void setReasonManualEncoding(int reasonManualEncoding) {
        this.reasonManualEncoding = reasonManualEncoding;
    }

    public int getReasonUsingVignette() {
        return reasonUsingVignette;
    }

    public void setReasonUsingVignette(int reasonUsingVignette) {
        this.reasonUsingVignette = reasonUsingVignette;
    }
}

File: src/main/kotlin/org/taktik/icure/services/external/rest/v1/dto/embed/IdentityDocumentReaderDto.kt
package org.taktik.icure.services.external.rest.v1.dto.embed

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.github.pozo.KotlinBuilder
import java.io.Serializable

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
@KotlinBuilder
data class IdentityDocumentReaderDto(
        val justificatifDocumentNumber: String? = null,
        val supportSerialNumber: String? = null,
        val timeReadingEIdDocument: Long? = null,
        val eidDocumentSupportType: Int = 0,
        val reasonManualEncoding: Int = 0,
        val reasonUsingVignette: Int = 0
) : Serializable
